### PR TITLE
Navigation settings

### PR DESCRIPTION
**Pull request by AnnBibyk**

### CODE REVIEW

File: Sources/Helpers/OAAppSettings.h [hunk 1]
```diff
@@ -271,6 +271,28 @@ typedef NS_ENUM(NSInteger, EOAMapMarkersMode)
 
 @end
 
+@interface OAProfileDrivingRegionConstant : OAProfileInteger
```
Comment (max-klaus) [inline Sources/Helpers/OAAppSettings.h:274]: OAProfileDrivingRegionConstant > OAProfileDrivingRegion

File: Sources/Helpers/OAAppSettings.h [hunk 2]
```diff
@@ -271,6 +271,28 @@ typedef NS_ENUM(NSInteger, EOAMapMarkersMode)
...
++ (instancetype) withKey:(NSString *)key defValue:(EOADrivingRegion)defValue;
+
+- (EOADrivingRegion) get;
+- (void) set:(EOADrivingRegion)drivingRegionConstant;
+- (EOADrivingRegion) get:(OAApplicationMode *)mode;
+- (void) set:(EOADrivingRegion)drivingRegionConstant mode:(OAApplicationMode *)mode;
+
+@end
+
+@interface OAMetricSystemConstant : OAProfileInteger
```
Comment (max-klaus) [inline Sources/Helpers/OAAppSettings.h:285]: OAMetricSystemConstant > OAMetricSystem

File: Sources/Helpers/OAAppSettings.h [hunk 3]
```diff
@@ -271,6 +271,28 @@ typedef NS_ENUM(NSInteger, EOAMapMarkersMode)
...
+
++ (instancetype) withKey:(NSString *)key defValue:(EOAMetricsConstant)defValue;
+
+- (EOAMetricsConstant) get;
+- (void) set:(EOAMetricsConstant)metricSystemConstant;
```
Comment (max-klaus) [inline Sources/Helpers/OAAppSettings.h:290]: metricSystemConstant > metricsConstant

File: Sources/Helpers/OAAppSettings.h [hunk 4]
```diff
@@ -271,6 +271,28 @@ typedef NS_ENUM(NSInteger, EOAMapMarkersMode)
...
+- (EOAMetricsConstant) get:(OAApplicationMode *)mode;
+- (void) set:(EOAMetricsConstant)metricSystemConstant mode:(OAApplicationMode *)mode;
```
Comment (max-klaus) [inline Sources/Helpers/OAAppSettings.h:292]: metricSystemConstant > metricsConstant

File: Sources/Controllers/2.0/Settings/NavigationSettings/OARouteParametersViewController.mm [hunk 5]
```diff
@@ -52,66 +58,153 @@ - (void) viewDidLoad
...
+        {
+            [parametersArr addObject:@{
+                @"type" : kCellTypeIconTitle,
+                @"title" : OALocalizedString(@"impassable_road"),
+                @"icon" : @"ic_custom_alert",
+                @"value" : @([self checkIfAnyParameterIsSelected:avoidParameters]),
+                @"key" : @"avoidRoads"
+            }];
+        }
+        for (auto& p : otherParameters)
```
Comment (PaulStets) [inline Sources/Controllers/2.0/Settings/NavigationSettings/OARouteParametersViewController.mm:124]: Add const.

**Pull request merged**